### PR TITLE
Can't set Start and End Dates in Salary Slip - 1

### PR DESCRIPTION
**Pull request by tundebabzy**
Reference - #9530, #9513 

Here is the first PR that creates Doctypes towards making creating Salary Slips and Process Payroll more flexible as well as fixing bugs like #9513 - Payroll Cycle, Payroll Period and Payroll Cycle Designation. Both Payroll Period and Payroll Cycle Designation are child tables of Payroll Cycle. I also added a test case. 

The target is that for each employee, ERPNext would check the Payment Cycle attached to the employee's designation to get pay period information needed for payroll runs.

### CODE REVIEW

**Comment by tundebabzy** [comment]
### Screenccast

peek 2017-06-30 23-33

**Comment by tundebabzy** [comment]
### Completed task 4
screenshot from 2017-07-01 13-43-40

File: erpnext/hr/doctype/payroll_cycle/test_payroll_cycle.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                "doctype": "Designation"}
+
+period = {"end_date": "2017-06-30", "start_date": "2017-06-01",
+          "doctype": "Payroll Period", "idx": "1"}
+
+cycle_dict = {"cycle_name": '_Test Payment Cycle All Employees', "company_name": '_Test Company',
+              "payroll_frequency": "Monthly", "doctype": "Payroll Cycle"}
+
+
+def get_document(dt, designatione):
```
Comment (rmehta) [inline erpnext/hr/doctype/payroll_cycle/test_payroll_cycle.py]: put this in `def setUp` method of the test case

File: erpnext/hr/doctype/payroll_cycle/test_payroll_cycle.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	if not frappe.db.exists(dt, designation.name):
+		designation.insert()
+	return designation
+
+get_document('Designation', designations)
+
+
+class TestPayrollCycle(unittest.TestCase):
+	def test_create_payment_cycle(self):
+		payroll_cycle = frappe.get_doc(cycle_dict)
```
Comment (rmehta) [inline erpnext/hr/doctype/payroll_cycle/test_payroll_cycle.py]: just add the dict here. its much more readable!

**Review by rmehta — CHANGES_REQUESTED** [review]

**Comment by tundebabzy** [comment]
Trying another approach

**Comment by magic-overflow** [comment]
Have you been working on another approach?

**Pull request closed without merging**